### PR TITLE
Responsive window simplification

### PR DESCRIPTION
**Pull request by DXCanas**
## Summary

Uses add/remove event listeners rather than relying on arrays & popping. 

Ensures that the right listener is removed

### CODE REVIEW

File: kolibri/core/assets/src/mixins/responsive-window.js [hunk 1]
```diff
@@ -147,56 +146,36 @@ function windowMetrics() {
...
-
-if (window.addEventListener) {
-  window.addEventListener('resize', windowResizeHandler, true);
-} else if (window.attachEvent) {
-  window.attachEvent('onresize', windowResizeHandler);
+  const listener = throttle(cb);
+  if (window.removeEventListener) {
+    window.addEventListener('resize', listener, true);
+  } else if (window.detachEvent) {
+    window.attachEvent('onresize', listener);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/mixins/responsive-window.js:163]: you're attaching _another_ event here :)

File: kolibri/core/assets/src/mixins/responsive-window.js [hunk 2]
```diff
@@ -147,56 +146,36 @@ function windowMetrics() {
...
-  windowListeners.pop(cb);
-}
-
-/* setup */
-
-if (window.addEventListener) {
-  window.addEventListener('resize', windowResizeHandler, true);
-} else if (window.attachEvent) {
-  window.attachEvent('onresize', windowResizeHandler);
+  const listener = throttle(cb);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/mixins/responsive-window.js:159]: You can't create a new instance of the the throttled callback in order to detach the old one: you'll need to detach the original instance.

(That was part of the motivation of the `windowListeners` array - keeping around a reference to the listeners that need to be removed).

File: kolibri/core/assets/src/mixins/responsive-window.js [hunk 3]
```diff
@@ -70,8 +70,7 @@
 */
 
 /* module internal state */
-
-const windowListeners = [];
+import { throttle } from 'frame-throttle';
```
Comment (indirectlylit) [inline kolibri/core/assets/src/mixins/responsive-window.js:73]: this import got moved to under the `/* module internal state */` comment. `prettier`?

File: kolibri/core/assets/src/mixins/responsive-window.js [hunk 4]
```diff
@@ -147,56 +146,36 @@ function windowMetrics() {
...
+  const listener = throttle(cb);
+  if (window.addEventListener) {
+    window.addEventListener('resize', listener, true);
+  } else if (window.attachEvent) {
+    window.attachEvent('onresize', listener);
+  }
 }
 
 function removeWindowListener(cb) {
-  windowListeners.pop(cb);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/mixins/responsive-window.js:163]: The bug here was that I used `pop`, thinking in a brain fart that it acted like python's `remove`.

The simplest fix would be to change this line to something like

```
_.remove(windowListeners, val => val === cb)
```

Comment (rtibbles) [reply kolibri/core/assets/src/mixins/responsive-window.js:163]: Or without lodash:
`windowListeners.splice(windowListeners.indexOf(cb), 1);`

Comment (DXCanas) [reply kolibri/core/assets/src/mixins/responsive-window.js:163]: I have a fix like this locally! This PR is mostly to see what you guys think

**Comment by DXCanas** [comment]
It's actually working for me as written, but I'll go ahead with the simple fix. Initially, just wanted to use the detach events. Got carried away. I think I'll include the simpler fix over on #1899

File: kolibri/core/assets/src/mixins/responsive-window.js (hunk 1 above)
Comment (DXCanas) [reply kolibri/core/assets/src/mixins/responsive-window.js:163]: oops

File: kolibri/core/assets/src/mixins/responsive-window.js (hunk 2 above)
Comment (DXCanas) [reply kolibri/core/assets/src/mixins/responsive-window.js:159]: Hm. I think I misunderstood the documentation, then: https://github.com/pelotoncycle/frame-throttle#use

Comment (DXCanas) [reply kolibri/core/assets/src/mixins/responsive-window.js:159]: ah I see. The docs use the same instance of the throttled listener. I'm creating a new one in the function. Got it

**Pull request closed without merging**